### ISSUE TITLE
Ubuntu/Debian md5sum on /usr/bin/dd-agent

### ISSUE THREAD

**Issue by gasiorroiasg** [DataDog/dd-agent#3207]
After proper install of datadog-agent from repository:
`https://apt.datadoghq.com/ stable main`

the file /usr/bin/dd-agent is reported on my hosts as modified.
I looked into that and I've found out that in postInst script 
a /usr/bin/dd-agent file is overwritten with symbolic link:
``
ln -sf $INSTALL_DIR/agent/agent.py `/usr/bin/dd-agent`
`` 
as a result it still has a md5 hash of overwritten file in package md5sums file.
and my periodic ``debsums -c``  check  reports an error on ``/usr/bin/dd-agent`` file

I consider it as a bug of install scripts and it would have been nice to get rid of this.

**Additional environment details (Operating System, Cloud provider, etc):**
i run Debian Wheezy,Debian Jessie hosts. some physical some virtual. 
install process on these don't differ and error still occur.

**Steps to reproduce the issue:**
1. install ``datadog-agent`` package on Debian Jessie host. run```apt-get install datadog-agent```
2. after installation complete run ```debsums -c```

**Additional information you deem important (e.g. issue happens only occasionally):**
The "bug" as i've called it does not affect overall datadog agent behavior or run but in my case i have to consider it as a minor security issue.

**Comment by abeluck** [comment]
I also encountered this today. It's definitely a bug in the packaging script.

The correct behavior would be to make the symlink in the `install` phase.

**Comment by olivielpeau** [comment]
Thanks @gasiorroiasg for reporting this!

TL;DR: this should be fixed in the next major version of the Agent (Datadog Agent 6: https://github.com/DataDog/datadog-agent) ; backporting the fix to Agent 5 while keeping full backward compatibility wouldn't be trivial so we're not going to do it.

We use [omnibus](https://github.com/chef/omnibus) to create the agent packages (project: https://github.com/DataDog/dd-agent-omnibus), and in some cases we're a bit limited by the capabilities that omnibus offers and the platforms we want to support.

AFAIK omnibus doesn't support defining symlinks directly in the package file when the symlink is outside of the main installation directory (in our case `/opt/datadog-agent`), so we can't define `/usr/bin/dd-agent` directly as a symlink in the package file list.

A workaround is to create the symlink in the postinst script (see https://github.com/DataDog/dd-agent-omnibus/blob/5.19.0/package-scripts/datadog-agent/postinst#L90), but unfortunately we also need to have the file present in the package file list to avoid an issue with the RPM package (see https://github.com/DataDog/dd-agent-omnibus/blob/5.19.0/config/projects/datadog-agent.rb#L149-L152)

So with the tooling we're using this is not such an easy issue to fix. We could work around that by only declaring the file in the file list of the package only for the RPM package ; and, for the DEB package, manage everything from the installation scripts (creating the symlink in `postinst`, and removing it in `postrm`).

Anyway, this will be fixed in the next major version of the Agent, so we're not going to work on this now.

I'll go ahead and close this, but let me know if you have any questions!
